internal/command/deploy/machines_tigrisstatics.go

```diff
@@ -0,0 +1,617 @@
+	}
+	if len(pubBytes) != 32 {
+		return "", fmt.Errorf("bad public key size: %d", len(pubBytes))
+	}
+
+	sct, err := box.SealAnonymous(nil, inputJson, (*[32]byte)(pubBytes), nil)
+	if err != nil {
+		return "", err
+	}
+
+	return base64.StdEncoding.EncodeToString(sct), nil
+}
+
+type headerInjectTransport struct {
+	transport http.RoundTripper
+	token     string
+	macaroon  string
+}
+
+func (t *headerInjectTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Add("Proxy-Tokenizer", t.token)
+	req.Header.Add("Proxy-Authorization", t.macaroon)
+	return t.transport.RoundTrip(req)
+}
+
+// Create the tigris bucket if not created.
+func (md *machineDeployment) staticsInitialize(ctx context.Context) error {
+
+	// TODO(allison): This is temporary debug code. Remove me.
+	tokenizerUrl := os.Getenv("FLY_TOKENIZER_URL")
+	if tokenizerUrl == "" {
+		return errors.New("please specify FLY_TOKENIZER_URL")
+	}
+	// TODO(allison): Pull this from somewhere sensible, such as an API on the Tokenizer directly?
+	tokenizerSealKey := os.Getenv("FLY_TOKENIZER_SEAL_KEY")
+	if tokenizerSealKey == "" {
+		return errors.New("please specify FLY_TOKENIZER_SEAL_KEY")
+	}
+
+	md.tigrisStatics.bucket = md.appConfig.AppName + "-statics"
+
+	if err := md.staticsEnsureBucketCreated(ctx, tokenizerSealKey); err != nil {
+		return err
+	}
+
+	// NOTE: This statics definition in the release sent to our API
+	//       should be correct and unmodified. *But*, because we're
+	//       modifying the app config in-place to ensure we don't have
+	//       double definitions for the static (both tigris & from local),
+	//       we'll pull an incorrect config if we grab it from machines.
+	//
+	// TODO(allison): We can probably solve this by sending the full statics config
+	//                to each machine as metadata and resynthesizing it during config save.
+	md.tigrisStatics.originalStatics = md.appConfig.Statics
+	md.appConfig.Statics = lo.Filter(md.appConfig.Statics, func(static appconfig.Static, _ int) bool {
+		return !staticIsCandidateForTigrisPush(static)
+	})
+
+	// TODO(allison): Temporary, while working on the POC for Tokenizer
+	encryptedToken, err := os.ReadFile("tokenized_key")
+	if err != nil {
+		return err
+	}
+
+	s3Config, err := awsconfig.LoadDefaultConfig(ctx,
+		awsconfig.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("dummy-access-key", "dummy-secret-key", "")),
+		awsconfig.WithRegion("auto"),
+	)
+	if err != nil {
+		return err
+	}
+	s3Config.BaseEndpoint = fly.Pointer(tigrisUrl)
+
+	parsedProxyUrl, err := url.Parse(tokenizerUrl)
+	if err != nil {
+		// Should be impossible, this is not runtime-controlled and issues would be caught before release.
+		return fmt.Errorf("could not parse tokenizer URL: %w", err)
+	}
+	s3HttpTransport := http.DefaultTransport.(*http.Transport).Clone()
+	s3HttpTransport.Proxy = http.ProxyURL(parsedProxyUrl)
+
+	cfg := flyconfig.FromContext(ctx)
+	userAuthHeader := cfg.Tokens.GraphQLHeader()
```

Comment:
This isn't guaranteed to give you macaroons, but should work for testing with your account. When you get things working, we can circle back to a more reliable way to get a macaroon for the request.